### DISCUSSION TITLE
Bash alternative for <<EOF ?

### DISCUSSION THREAD

**Discussion by mihail-milev** [nushell/nushell#13366] [Q&A]
Hi all,

I am used to do stdin pipe in bash, e.g.:

```
$ cat <<EOF > some_file.txt
> Hello, world!
> EOF
```

Which will save "Hello, world!" to some_file.txt. I tried doing this in nushell with:

```
~> cat <i EOF o> some_file.txt
```

and some other variants, but none worked.

Is this supported? How can I do a similar trick?

BR
M

**Comment by ghost** [comment]
the easiest solution is to just use quotes (single, double, or backtick) - all of them allow you to just press return and continuing them on another line:
```nu
"Hello, world!" | save some_file.txt

'One line
another line' | save some_file.txt

`One line
another line` | save some_file.txt
```

since this can look ugly in script which are indented i personally like to use arrays for multi-line strings:

```nu
if true {
  [
    "One line"
    "Another line"
  ] | str join "\n" | save some_file.txt
}
```

altough you can also use `\n` in a string:

```nu
"One line\nAnother line" | save some_file.txt
```

none if these is equivalent, but i personally vastly prefer them since it is a lot more readable (especially in more complex locations like
```bash
curl -F google.com <<EOF 2>>/dev/null | jq .foo | sed 's/foo/bar/g' >> foo.txt`
multiline curl
form data
EOF
```
)

**Reply by mihail-milev** [reply]
Hi,

thank you for suggesting these. Yes, I agree that these are possibilities, but they don't do exactly the same. When using the `<<EOF` I don't need to escape characters, while with pipes I have to. Copying a large text into the console has to then happen through a file, or at least through some replace-all/preprocessing procedure.

BR
M

**Reply by ghost** [reply]
for the usecase of pasting stuff into terminal i just use a clipboard-cli-tool, which just dumps it directly into the terminal like <https://github.com/Slackadays/Clipboard> or xclip (other options: <https://wiki.archlinux.org/title/Clipboard>)

with that you can independent of your shell just `cb paste | whatever` 

and for the usecase of writing your clipboard to a file: i just use a text-editor

**Comment by weirdan** [comment]
Nushell now has raw strings (see https://www.nushell.sh/book/working_with_strings.html#raw-strings) which may help here

```nushell
r#'
Hello, world!
'# | save some_file.txt
```
